### ISSUE TITLE
`no-undefined-types` - false positive on `@link` to symbol declared inside `declare global` block defined in another module

### ISSUE THREAD

**Issue by DetachHead** [gajus/eslint-plugin-jsdoc#1708]
this is the same issue as #1701, but it's still broken when the globally declared symbol is defined in a separate module.

## Expected behavior
no error

## Actual behavior
```
The type 'foo' is undefined. (jsdoc/no-undefined-types)
```

## ESLint Config

```js
/** @type {import('eslint').Linter.Config} */
const config = {
  extends: [
    'plugin:@typescript-eslint/recommended',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module',
    project: './tsconfig.json',
  },
  rules: {
    'import/no-undefined-types': 'error',
  },
};

module.exports = config;
```

## ESLint sample

```ts
// foo.ts
declare global {
  const foo: number;
}
```

```ts
// bar.ts
/**
 * {@link foo}
 */
export const bar = 1;
```

typescript is still able to recognize the symbol when it's defined in a separate file

## Environment

- Node version: 22.17.0
- ESLint version 8.57.1
- `eslint-plugin-jsdoc` version: 63.0.5

**Comment by brettz9** [comment]
Since ESLint only works at the level of individual files, we cannot practically resolve this. You'd have to use the `definedTypes` option or possibly `/* globals */`.

**Comment by brettz9** [comment]
Closing but feel free to comment further as needed.

**Comment by DetachHead** [comment]
can it use the typescript project config from my configured `tsconfig.json` to determine all the included files?

**Comment by brettz9** [comment]
Ours is just a plugin, and there is no ESLint API we can use to query ESLint to do file-reading of our choosing. We don't want to reimplement all of the file-reading that ESLint does for us (or would ideally do for us).
